Sprint 6/Module 2-Custom Controls/UIControl/UIControl/CustomControl.swift

```diff
@@ -0,0 +1,116 @@
+            //spacing
+            let space: CGFloat = (componentDimension * count) + (8.0 * count)
+            //create labels, set label's frame to size
+            let label = UILabel(frame: CGRect(x: space, y: 0, width: componentDimension, height: componentDimension))
+            //add one as subview
+                self.addSubview(label)
+            //store label in local array with append
+            labels.append(label)
+            //add tag for each view
+            label.tag = number
+            count += 1.0
+            //set font to bold system font size 32
+            label.font = UIFont.boldSystemFont(ofSize: 32.0)
+            //set text to Unicode star
+            label.text = "✭"
+            //alignment center for label
+            label.textAlignment = .center
+            //set textColor to active for the first or inactive for the others
+            label.textColor = componentInactiveColor
+        }
+    }
+    override var intrinsicContentSize: CGSize {
+        let componentsWidth = CGFloat(componentCount) * componentDimension
+        let componentsSpacing = CGFloat(componentCount + 1) * 8.0
+        let width = componentsWidth + componentsSpacing
+        return CGSize(width: width, height: componentDimension)
+    }
+    
+    //Mark: - Touch handlers
+    override func beginTracking(_ touch: UITouch, with event: UIEvent?) -> Bool {
+        updateValue(at: touch)
+        return true
+    }
+    
+    override func continueTracking(_ touch: UITouch, with event: UIEvent?) -> Bool {
+        let touchPoint = touch.location(in: self)
+        if bounds.contains(touchPoint) {
+            sendActions(for: [.valueChanged, .touchDragInside])
+            updateValue(at: touch)
+        } else {
+            sendActions(for: [.touchDragOutside])
+        }
+        return true
+    }
+    
+    override func endTracking(_ touch: UITouch?, with event: UIEvent?) {
+        defer { super.endTracking(touch, with: event) }
+        guard let touch = touch else { return }
+        let touchPoint = touch.location(in: self)
+        
+        if bounds.contains(touchPoint) {
+            sendActions(for: [.touchUpInside, .valueChanged])
+        } else {
+            sendActions(for: [.touchUpOutside])
+        }
+    }
+    
+    override func cancelTracking(with event: UIEvent?) {
+        sendActions(for: [.touchCancel])
+    }
+    
+    func updateValue(at touch: UITouch) {
+        //loop through component labels and detects whether touch location is contained in each label's frame
+        for label in labels {
+            let touchPoint = touch.location(in: self)
+            //when touch overlaps label set control's value to that tag
+            //update label colors to reflect current touch send valueChanged action
+            if label.frame.contains(touchPoint) {
+                value = label.tag
+                sendActions(for: [.valueChanged])
+                label.performFlare()
```

Comment:
If label and touch are at same location you need to make it's color active
```suggestion
                label.textColor = componentActiveColor
                label.performFlare()
```